### DISCUSSION TITLE
Save current settings as preset

### DISCUSSION THREAD

**Discussion by PascalHaury** [lllyasviel/Fooocus#1782] [Ideas]
Hey, 

it would be great if we could export/save the current selected Settings/Styles/Models as (for windows) a new bat file to start with the same settings. Something like the export/import workflow like in ComfyUI.

I experiment many times and when i find a combo that suits me, it would be great if we could save all settigs instead of manually creating a new preset, which can get quite annoying.

A simple button like "Save as Preset" would be nice. A import or load preset button would also be nice, then we wouldn't have to restart Fooocus when we want/need to change to a new preset

**Comment by JT-Gresham** [comment]
You can make custom presets...just look in your presets folder and make a copy of the one you are using. Rename the file to what you want the preset to be, then edit it with your setup....easy.  Then load up Fooocus with your --preset....done.

**Reply by PascalHaury** [reply]
I'm doing this already, like i wrote above. But i think it would be a nice QoL improvement, if we could skip all the maual steps. I have at the moment already about 50 different presets. It works, but could be more user friendly and integrated in Fooocus

**Reply by JT-Gresham** [reply]
Oh..my bad.  I gotcha and agree...an easier way to save/swap between configs would be awesome.  My apologies for my brain not registering the whole post...lol  I blame COVID..

**Reply by mashb1t** [reply]
The feature is in the backlog https://github.com/lllyasviel/Fooocus/issues/676, but please check out if https://github.com/lllyasviel/Fooocus/pull/1570 fulfills your needs, at least for the loading part.

**Reply by metercai** [reply]
The derivative version of Fooocus, SimpleSDXL already has this feature. May be tried it before #1570 launch. https://github.com/metercai/SimpleSDXL

**Comment by TemplarIRL** [comment]
I would just like to piggyback this QoL suggestion with a request for a better system for viewing models, in general.
While I don't mind a dropdown, am image tile for quick reference with a short description would be ideal. Kind of like civitai... 
I've seen it with A1111 and would swap simply for that feature, but prefer Fooocus's overall feel.
When you have 3 or 4 checkpoints or a dozen LoRa with similar names it becomes a chore to remember which is which.

**Comment by Cad5z5** [comment]
I came here to make this exact same suggestion as OP. I am glad, I did a search before posting. Just as OP said, A simple button like "Save as Preset" would be nice quality of life upgrade. Also the reason I often need it is because of random crashes on linux. After long session the whole system freezes and crashes or logs out user, I guess it runs out of memory. Just starting up with the same settings would speed stings up so much more. Otherwise If you manually set up 15 settings, it takes a minute or two. Loading a saved preset and saving preset on exit and loading previous on startup would be so productive.

**Reply by mashb1t** [reply]
you can already reload almost all settings (except img2img) from image metadata by activating it in the config file.

**Reply by ILikeToasters** [reply]
I did create a tool I use for creating presets easily using metadata from the log and is very easy and quick. https://ai.kleebz.com/fooocus/presets/

**Reply by DavidDragonsage** [reply]
It would be wonderful if this great new tool could eventually be integrated into Fooocus itself.
